search/ranker.cpp

```diff
@@ -115,8 +123,20 @@ NameScores GetNameScores(FeatureType & ft, Geocoder::Params const & params,
       UpdateNameScores(vec, lang, slice, bestScores);
       UpdateNameScores(vec, lang, sliceNoCategories, bestScores);
 
+      /// @todo
+      /// 1. Make sure that this conversion also happens for Address and POI results,
+      /// where street is only one component.
+      /// 2. Make an optimization: If there are no synonyms or "strasse", skip this step.
       if (type == Model::TYPE_STREET)
       {
+        // Searching for "Santa Fe" should rank "Avenida Satna Fe" like FULL_MATCH or FULL_PREFIX, but not SUBSTRING.
+        {
+          TokensVector cleaned(RemoveStreetSynonyms(vec.GetTokens()));
+          UpdateNameScores(cleaned, lang, slice, bestScores);
+          UpdateNameScores(cleaned, lang, sliceNoCategories, bestScores);
+        }
+
+        /// @todo Should definitely add "platz"-"pl". And maybe "gasse"-"g", "allee"-"al"?
```

Comment:
```
✗ rg -iwc 'str\.' all.txt       
13863

✗ rg -iwc 'pl\.' sorted.txt 
1451

✗ rg -iwc ' g\.' sorted.txt
1217


✗ rg -iwc ул\. all.txt
32450

✗ rg -iwc пл\. all.txt
2600

✗ rg -iwc пер\. all.txt
5601

```

Some interesting addresses (check yourself):
```
E.-Claudius-Str./H.-Mann-Allee
Bernauer Str./Wassersportvereine
```

https://cdn.organicmaps.app/apk/unique_osm_names.zip

Reply:
The patch is for German-lang addresses that have long, short and concatenated prefix.
strasse, platz is enough